### DISCUSSION TITLE
Services no longer functions, after enforcing fromPromise...

### DISCUSSION THREAD

**Discussion by thetumper** [statelyai/xstate#4611] [Q&A]
Trying to get the major upgrade working.  My usages was with async service functions, provided to the services property.  Doing so, I could swap in a set of dummy functions for unit tests, and mock them with jest.  Now that each function has to be wrapped with fromPromise, its become an ActorLogic, not a function.  No longer can I mock this with jest.

This change to everything being Actors and ActorLogics may make sense to some of you -- but I am really struggling with the move from services being javascript functions, to ActorLogics.  It is messing up things alot.  Any tips?  What am I missing?

**Comment by Andarist** [comment]
You should still be able to mock things with dummy `fromPromise(() => Promise.resolve('whatever'))`. At the mocking level, this is still the same - you replace one value with another. Previously you could replace an async function with another async function and today you can replace an actor logic with another actor logic.

**Reply by thetumper** [reply]
Yes, it can replace with another actor logic.  But, what I am needing to do, is use jest.spyOn, with assertions about whether the spied function was called or not, and sometimes, checking the args...in some case, it should be called with "a", other case, with "b".  But, now it is not a function, and jest does not spy on ActorLogics.  So how can I do it?

**Reply by Andarist** [reply]
I'm not sure which jest API are you using here - since they have a couple that relate to spying/mocking/etc. If you could share a piece of your code that would help me to answer this.
